### DISCUSSION TITLE
Is it possible to intercept HTTP connections in Rust side?

### DISCUSSION THREAD

**Discussion by erayerdin** [tauri-apps/tauri#4690] [Q&A] [answered]
It is possible to intercept HTTP connections on Javascript side. Is it possible to intercept the outgoing and incoming HTTP connections on the Rust side in Tauri project? If so, would you redirect me to the relevant code example or documentation?

Thanks in advance.

**Comment by w-henderson** [comment, marked as answer]
How do you mean "intercept"? You could write a wrapper around `fetch` requests on the JavaScript side or create a Tauri command to handle things on the Rust side. If you mean actually having some event that fires when the front-end makes an HTTP request which allows for the manipulation of its contents and response, this is not possible as HTTP requests are handled by the WebView runtime and not by Rust AFAIK.

**Reply by erayerdin** [reply]
What I'm trying to do is to block the client to connect to some IP addresses, not mutate it actually, but if it is handled by the runtime webview, then it's not possible I guess.

**Reply by w-henderson** [reply]
In what context is the client able to choose which IP address to connect to? Is it like an "import photo from URL" sort of thing?

**Reply by nothingismagick** [reply]
This is what the CSP setting is for - if you know it at build time, you can prevent the webview from connecting to specific URLs.
